electionleaflets/apps/leaflets/models.py

```diff
@@ -92,24 +93,62 @@ def save(self, *args, **kwargs):
 
 
 class LeafletImage(models.Model):
+    ORIENTATION_CHOICES = (
+        (1, 'Horizontal (normal)'),
+        (2, 'Mirror horizontal'),
+        (3, 'Rotate 180'),
+        (4, 'Mirror vertical'),
+        (5, 'Mirror horizontal and rotate 270 CW'),
+        (6, 'Rotate 90 CW'),
+        (7, 'Mirror horizontal and rotate 90 CW'),
+        (8, 'Rotate 270 CW'),
+    )
+
     leaflet = models.ForeignKey(Leaflet, related_name='images')
     image = ImageField(upload_to="leaflets", max_length=255)
     raw_image = ImageField(upload_to="raw_leaflets", blank=True, max_length=255)
     legacy_image_key = models.CharField(max_length=255, blank=True)
     image_type = models.CharField(choices=constants.IMAGE_TYPES,
         null=True, blank=True, max_length=255)
     image_text = models.TextField(blank=True)
+    orientation = models.PositiveSmallIntegerField(choices=ORIENTATION_CHOICES, default=1)
+    exif_data = models.BinaryField(blank=True)
 
     class Meta:
         ordering = ['image_type']
 
+    def _remove_exif_data(self):
+        full_image = Image.open(self.image)
+
+        if full_image.info.get('exif'):
+            exif_dict = piexif.load(full_image.info['exif'])
```

Comment:
There are two thoughts to this:

- why would we want the exif data, which is largely about camera and lens types and not related to election leaflets
- A problem occurs because the data is a binary blob - it can be extracted into a dict, but not then stored easily as JSON as there is binary data included in the exif data, and we can never be entirely sure what is useful string data and what can be thrown away